File: sprint1/src/main/java/com/c0821g1/sprint1/service/CustomerService.java
package com.c0821g1.sprint1.service;

import com.c0821g1.sprint1.entity.customer.Customer;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;
import java.util.Optional;

public interface CustomerService {
    //    VyLTT - customer list
    Page<Customer> getAll(Pageable pageable);
    List<Customer> getAll1();

    Optional<Customer> findById(Integer id);

    //    VyLTT - search by name, email, phone, identify number
    Page<Customer> findCustomerByNameAndEmailAndPhoneAndIdentify(Pageable pageable,
                                                                 String customer_name,
                                                                 String customer_email,
                                                                 String customer_phone,
                                                                 String customer_identify_number);
    void deleteCustomer(Integer id);
}

File: c0821g1-building-management/src/app/service/customer/customer.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs';
import {Customer} from '../../model/customer/customer';

@Injectable({
  providedIn: 'root'
})
export class CustomerService {
  // VyLTT
  private API_URL = 'http://localhost:8080/api/customer';

  constructor(private httpClient: HttpClient) {
  }

  // VyLTT - get by id
  getCustomerById(customerId: number): Observable<any> {
    return this.httpClient.get<Customer>(this.API_URL + '/' + customerId);
  }

  // VyLTT - get list customer
  getAllCustomer(): Observable<Customer[]> {
    return this.httpClient.get<Customer[]>(this.API_URL + '/list-select');
  }

  // VyLTT - search customer by name, identify number, email, phone
  search(page, customerName, customerIdentifyNumber, customerPhone, customerEmail): Observable<any> {
    return this.httpClient.get(this.API_URL + '/search?page=' + page +
      '&customer_name=' + customerName +
      '&customer_identify_number=' + customerIdentifyNumber +
      '&customer_phone=' + customerPhone +
      '&customer_email=' + customerEmail);
  }

  // VyLTT - delete customer
  deleteCustomerById(deleteId: number): Observable<any> {
    return this.httpClient.delete<void>(this.API_URL + '/delete/' + deleteId);
  }
}
